mixer/cmd/build.go

```diff
@@ -199,113 +213,163 @@ var buildFormatNewCmd = &cobra.Command{
 		if err != nil {
 			fail(err)
 		}
+		// Update mixer to build version +20, to build the bundles with +20 inside not +10
 		if err = b.UpdateMixVer(ver + 20); err != nil {
 			failf("Couldn't update Mix Version")
 		}
 
-		// Set format to format+1 so that the format file inserted into the
-		// update content is the new one
-		newFormat, err := strconv.Atoi(b.State.Mix.Format)
-		if err != nil {
-			fail(err)
-		}
-		newFormat++
-		if err = b.UpdateFormatVersion(strconv.Itoa(newFormat)); err != nil {
+		// Change the format internally just for building bundles so the right
+		// /usr/share/defaults/swupd/format is inserted; preserve old format
+		oldFormat := b.State.Mix.Format
+		b.State.Mix.Format = buildFlags.newFormat
+
+		// Build bundles normally. At this point the bundles to be deleted should still
+		// be part of the mixbundles list and the groups.ini
+		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
 			fail(err)
 		}
-		setWorkers(b)
 
-		fmt.Println(" Backing up full groups.ini")
-		// Back up groups.ini in case we have deprecated bundles to delete
-		if err = b.CopyFullGroupsINI(); err != nil {
+		// Wipes the <bundle>-info files and replaces them with
+		// <bundle>/ directories that are empty. When the manifest creation happens it will
+		// mark all files in that bundles as deleted.
+		f := func(filename string) error {
+			bdir := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer, filename)
+			// This bundle is deprecated, remove the info file
+			if err = os.RemoveAll(bdir + "-info"); err != nil {
```

Comment:
maybe just `os.Remove`?

* bundle `foo` is deprecated
* bundle `foo-info` is not
* bundle `foo-info` has a chroot to provide the content instead of a `foo-info-info` file.
* bundle `foo` *also* has a chroot to provide the content instead of a `foo-info` file.
* this will recursively remove the `foo-info` chroot instead of the `foo` chroot.

We should just use `os.Remove` so we don't accidentally remove bundle content we don't intend to. There's probably some other sanity checking that needs to happen, like checking to make sure `foo-info` isn't a bundle itself.

note that `mixer build update` will fail if there is a `foo-info` and a `foo` bundle that both use a chroot instead of an `info` file, so this isn't a huge deal, but still definitely don't want to be removing that content on accident.

Reply:
bump about the `os.Remove` bug I mentioned in my previous review.